include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -656,26 +658,29 @@ struct __result_and_scratch_storage : __result_and_scratch_storage_base
 #endif
     }
 
-    bool
-    is_USM() const
+    _T
+    __wait_and_get_value(sycl::event __event) const
     {
-        return __supports_USM_device;
+        if (is_USM())
+            __event.wait_and_throw();
+
+        return __get_value();
     }
 
     // Note: this member function assumes the result is *ready*, since the __future has already
     // waited on the relevant event.
     _T
     __get_value(size_t idx = 0) const
     {
-        assert(idx < __result_n);
+        assert(__result_n == 1);
         if (__use_USM_host && __supports_USM_device)
         {
-            return *(__result_buf.get() + idx);
+            return *(__result_buf.get());
         }
         else if (__supports_USM_device)
         {
             _T __tmp;
-            __exec.queue().memcpy(&__tmp, __scratch_buf.get() + __scratch_n + idx, 1 * sizeof(_T)).wait();
+            __exec.queue().memcpy(&__tmp, __scratch_buf.get() + __scratch_n, 1 * sizeof(_T)).wait();
```

Comment:
It doesn't make sense to make these changes within `__get_value` and revert the other ones (private and idx param).  I don't think these changes are required for this PR, and we have intentions to do a real refactor of this struct and and `__future` in the subsequent release, so lets just leave this alone for now.

Reply:
done